lib/serviceClient.ts

```diff
@@ -240,12 +240,24 @@ export class ServiceClient {
     let httpPipeline: RequestPolicy = this._httpClient;
     if (this._requestPolicyFactories && this._requestPolicyFactories.length > 0) {
       for (let i = this._requestPolicyFactories.length - 1; i >= 0; --i) {
-        httpPipeline = this._requestPolicyFactories[i].create(
+        const nextPolicyPipeline = this._requestPolicyFactories[i].create(
           httpPipeline,
           this._requestPolicyOptions
         );
+
+        if (httpRequest.redirectLimit !== undefined && nextPolicyPipeline instanceof RedirectPolicy) {
```

Comment:
I don't understand this if block -- it looks like you're just skipping any redirect policies that are already in the pipeline?

Reply:
You can only have one redirect policy in the pipeline. If you have more than one, the first one will execute the redirects and any subsequent ones would never see a response with a redirect status code.

Reply:
Also note: 
>It turns out the existing redirectPolicy was never executed as the call to node-fetch does not override the default value of redirect='follow'. As it was, node-fetch processed all redirects before the redirectPolicy was executed. I have modified this so that if a redirectLimit is defined, the call to node-fetch will use redirect='manual'

Reply:
It feels strange to enforce this dynamically, before executing each policy. I get that _requestPolicyFactories is opaque, but I don't think the bad design of the policy factories excuses creating further bad design in `ServiceClient`.

In general it feels bad to strongly couple knowledge of a particular pipeline policy into `sendRequest`.

Reply:
As above, it's obvious now that I should handle this in the redirectPolicy pipeline.